### ISSUE TITLE
Unexpected rvalue type in eval attribute with quantities

### ISSUE THREAD

**Issue by cmft** [taurus-org/taurus#433]
The EvalAttribute when the URI explicit defines a Quantity, e.g. "eval:Q('1m')", the  rvalue type returns a `numpy.ndarray` instead of scalar (int/float), although the attribute data_format says it is a scalar.

You can reproduce it with this code:
```
import taurus
a = taurus.Attribute("eval:Q('1m')")
a.data_format
Out[14]: 0 # SCALAR
type(a.read().rvalue.magnitude)
Out[15]: numpy.ndarray
```

**Comment by cmft** [comment]
The  bug is in `evalAttribute applyTransformation` method.  The uses of `numpy.shape` method over a Quantity mutate the Quantity's magnitude type.

```
In [1]: import numpy

In [2]: from taurus.external.pint import Quantity

In [3]: v = Quantity(1)

In [4]: print type(v.magnitude)
<type 'int'>

In [5]: value_dimension = len(numpy.shape(v))

In [6]: print type(v.magnitude)
<type 'numpy.ndarray'>
```

**Comment by cmft** [comment]
I opened an issue in pint [#509](https://github.com/hgrecco/pint/issues/509)

**Comment by cpascual** [comment]
fixed in #448
